# Analysis

## Imports

In [70]:
import pandas as pd
import scipy.stats as stats

Imports required libraries for this notebook

## Data

In [71]:
mergedDataLocation = r"ModifiedData/ExportedData.csv"
newMergedOutput = r"ModifiedData/GraphingData.csv"

Creates an easily editable variable with the location of the cleaned and merged dataset

In [72]:
MergedDF = pd.read_csv(mergedDataLocation, index_col=0)
MergedDF

,Country name,Year,Code,"GDP per capita, PPP (constant 2021 international $)",Life evaluation (3-year average),Rank,Lower whisker,Upper whisker,Explained by: Log GDP per capita,Explained by: Social support,Explained by: Healthy life expectancy,Explained by: Freedom to make life choices,Explained by: Generosity,Explained by: Perceptions of corruption,Dystopia + residual
0,Finland,2023,FIN,57063.5230,7.741,1,7.667,7.815,1.844,1.572,0.695,0.859,0.142,0.546,2.082
1,Denmark,2023,DNK,72097.3050,7.583,2,7.500,7.665,1.908,1.520,0.699,0.823,0.204,0.548,1.881
2,Iceland,2023,ISL,67255.6700,7.525,3,7.433,7.618,1.881,1.617,0.718,0.819,0.258,0.182,2.050
3,Sweden,2023,SWE,63114.6800,7.344,4,7.267,7.422,1.878,1.501,0.724,0.838,0.221,0.524,1.658
4,Israel,2023,ISR,48432.1100,7.341,5,7.277,7.405,1.803,1.513,0.740,0.641,0.153,0.193,2.298
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1760,Burundi,2011,BDI,951.1885,3.678,152,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1761,Sierra Leone,2011,SLE,2471.7185,3.586,153,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1762,Central African Republic,2011,CAF,1517.1484,3.568,154,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1763,Benin,2011,BEN,2732.0125,3.493,155,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


Loads the merged data into a Pandas DataFrame object and displays it

## Analysis

### Generate List-wise Data for linear modelling

In [73]:
EconomicData = list(MergedDF["GDP per capita, PPP (constant 2021 international $)"])
HappyData = list(MergedDF["Life evaluation (3-year average)"])

pd.DataFrame({"GDP per capita":EconomicData,"Reported Happiness Score":HappyData})

,GDP per capita,Reported Happiness Score
0,57063.5230,7.741
1,72097.3050,7.583
2,67255.6700,7.525
3,63114.6800,7.344
4,48432.1100,7.341
...,...,...
1760,951.1885,3.678
1761,2471.7185,3.586
1762,1517.1484,3.568
1763,2732.0125,3.493


Creates two lists, one containing GDP per Capita and the other containing the Reported Happiness Score, from the Merged Dataframe. The lists are then temporarily put into a new dataframe to be displayed with headers, for comparison against the larger DataFrame.

### Model Creation

In [74]:
# As shown at https://how-to-data.org/how-to-fit-a-linear-model-to-two-columns-of-data/
model = stats.linregress(EconomicData, HappyData)

f"Model equation: y = {model.slope}x + {model.intercept}; a standard error of {model.stderr}, p-value of {model.pvalue}, and R-squared of {model.rvalue**2}"

'Model equation: y = 3.266360649187537e-05x + 4.637895887823021; a standard error of 7.186079132766014e-07, p-value of 3.0728336671769754e-299, and R-squared of 0.5395744532232545'

Performs a Linear Regression and Outputs the line generated

### Prediction Generation

In [75]:
MergedDF["Predicted Happiness"] = MergedDF["GDP per capita, PPP (constant 2021 international $)"].apply(lambda x : (model.slope * x) + model.intercept)

MergedDF

,Country name,Year,Code,"GDP per capita, PPP (constant 2021 international $)",Life evaluation (3-year average),Rank,Lower whisker,Upper whisker,Explained by: Log GDP per capita,Explained by: Social support,Explained by: Healthy life expectancy,Explained by: Freedom to make life choices,Explained by: Generosity,Explained by: Perceptions of corruption,Dystopia + residual,Predicted Happiness
0,Finland,2023,FIN,57063.5230,7.741,1,7.667,7.815,1.844,1.572,0.695,0.859,0.142,0.546,2.082,6.501796
1,Denmark,2023,DNK,72097.3050,7.583,2,7.500,7.665,1.908,1.520,0.699,0.823,0.204,0.548,1.881,6.992854
2,Iceland,2023,ISL,67255.6700,7.525,3,7.433,7.618,1.881,1.617,0.718,0.819,0.258,0.182,2.050,6.834709
3,Sweden,2023,SWE,63114.6800,7.344,4,7.267,7.422,1.878,1.501,0.724,0.838,0.221,0.524,1.658,6.699449
4,Israel,2023,ISR,48432.1100,7.341,5,7.277,7.405,1.803,1.513,0.740,0.641,0.153,0.193,2.298,6.219863
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1760,Burundi,2011,BDI,951.1885,3.678,152,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,4.668965
1761,Sierra Leone,2011,SLE,2471.7185,3.586,153,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,4.718631
1762,Central African Republic,2011,CAF,1517.1484,3.568,154,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,4.687451
1763,Benin,2011,BEN,2732.0125,3.493,155,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,4.727133


Adds a Prediction column to the MergedDF for graphing

### Get Covariance and correlation of Per Capita GDP and Happiness Score

In [76]:
gdpHappinessCovariance = MergedDF["GDP per capita, PPP (constant 2021 international $)"].cov(MergedDF["Life evaluation (3-year average)"])
gdpHappinessCorrelation = MergedDF["GDP per capita, PPP (constant 2021 international $)"].corr(MergedDF["Life evaluation (3-year average)"])
f"GDP per Capita and Reported Happiness Score have a covariance of {gdpHappinessCovariance} and a correlation of {gdpHappinessCorrelation}"

'GDP per Capita and Reported Happiness Score have a covariance of 20527.554068969566 and a correlation of 0.7345573178610736'

Gets correlation and covariance of GDP per Capita and Happiness Score, showing a strong positive correlation of 0.7345573178610736

## Save Data for Graphing

In [77]:
MergedDF.to_csv(newMergedOutput)

Saves data to be used in the graphing notebook